client/client.go

```diff
@@ -18,6 +18,8 @@ type Client interface {
 	BulkIndexClose(context.Context) error
 	Checker(ctx context.Context, state *health.CheckState) error
 	CreateIndex(ctx context.Context, indexName string, indexSettings []byte) error
+	DeleteDocument(ctx context.Context, index, documentID string) error
+	DeleteDocumentByQuery(ctx context.Context, index string, query map[string]interface{}) error
```

Comment:
See my original comment on this. I've tried to keep this function consistent with the existing Search function's signature. We really shouldn't be using `interface{}` params in interfaces, especially when unnecessary
```suggestion
	DeleteDocumentsBySearch(ctx context.Context, search Search) error
```
Note, the `Search` type includes a header with the index name in it hence why there's no `index` param.